tools/topology/topology2/include/platform/intel/dmic-generic.conf

```diff
@@ -0,0 +1,196 @@
+
+	# Stream name for the host widget for the DMIC0 pipeline. This should be the same as the
+	# PCM capabilities for the PCM that this widget is connected to
+	DMIC0_HOST_STREAM_NAME	"Highpass Capture 6"
+
+	# Stream name for the host widget for the DMIC0 pipeline. This should be the same as the
+	# PCM capabilities for the PCM that this widget is connected to
+	DMIC1_HOST_STREAM_NAME	"Highpass Capture 7"
+
+	# DMIC0 EQIIR byte control name
+	DMIC0_EQIIR_BYTES_NAME "eqiir_bytes_10"
+
+	# DMIC1 EQIIR byte control name
+	DMIC1_EQIIR_BYTES_NAME "eqiir_bytes_11"
+
+	CHANNELS	2
+
+	FORMAT "s32le"
+}
+
+# DMIC DAI definition
+Object.Dai {
+	DMIC.0 {
+		name dmic01
+		id $DMIC0_ID
+		default_hw_config_id $DMIC0_ID
+		unmute_ramp_time_ms 200
+		num_pdm_active $NUM_PDM
+		IncludeByKey.DMIC_CH_COUNT {
+			"[1-2]" "include/platform/intel/pdm-config-2ch.conf"
+			"[3-4]" "include/platform/intel/pdm-config-4ch.conf"
+		}
+		Object.Base.hw_config.DMIC0 {}
+		Object.Widget.dai.0 {
+			stream_name dmic01
+			index $DMIC0_PIPELINE_ID
+			type dai_out
+			period_sink_count 2
+			period_source_count 0
+			format $FORMAT
+		}
+	}
+	DMIC.1 {
+		name dmic16k
+		id $DMIC1_ID
+		default_hw_config_id $DMIC1_ID
+		sample_rate 16000
+		num_pdm_active $NUM_PDM
+		IncludeByKey.DMIC_CH_COUNT {
+			"[1-2]" "include/platform/intel/pdm-config-2ch.conf"
+			"[3-4]" "include/platform/intel/pdm-config-4ch.conf"
+		}
+		Object.Base.hw_config.DMIC1 {}
+		Object.Widget.dai.0 {
+			stream_name dmic16k
+			index $DMIC1_PIPELINE_ID
+			type dai_out
+			period_sink_count 2
+			period_source_count 0
+			format $FORMAT
+		}
+	}
+}
+
+# Capture pipeline definition
+Object.Pipeline {
+	eq-iir-volume-capture.100 {
+		dynamic_pipeline	$DYNAMIC_PIPELINE
+		index $DMIC0_PIPELINE_ID
+		Object.Widget.pipeline.1 {
+			stream_name 'dai.DMIC.0.capture'
+		}
+		Object.Widget.host.capture {
+			stream_name $DMIC0_HOST_STREAM_NAME
+		}
+		Object.Widget.pga.1 {
+			Object.Control.mixer.1 {
+				name 'Capture Volume'
+			}
+			Object.Control.mixer.2 {
+				mute_led_use	1
+				mute_led_direction 1
+				name 'Capture Switch'
+			}
+		}
+		Object.Widget.eqiir.1 {
+			Object.Control.bytes.1 {
+				name $DMIC0_EQIIR_BYTES_NAME
+				IncludeByKey.DMIC0_EQIIR_COEFF {
+					"highpass_40hz_0db_48khz" "include/components/eqiir/highpass_40hz_0db_48khz.conf"
+					"highpass_40hz_20db_48khz" "include/components/eqiir/highpass_40hz_20db_48khz.conf"
+				}
+			}
+		}
+		format $FORMAT
+		channels	$CHANNELS
+	}
+	eq-iir-volume-capture.101 {
+		Object.Widget.pipeline.1 {
+			stream_name 'dai.DMIC.1.capture'
+		}
+		Object.Widget.host.capture {
+			stream_name $DMIC1_HOST_STREAM_NAME
+		}
+		Object.Widget.pga.1 {
+			Object.Control.mixer.1 {
+				name '2nd Capture Volume'
+			}
+		}
+		Object.Widget.eqiir.1 {
+			Object.Control.bytes.1 {
+				name $DMIC1_EQIIR_BYTES_NAME
+				IncludeByKey.DMIC1_EQIIR_COEFF {
+					"highpass_40hz_20db_16khz" "include/components/eqiir/highpass_40hz_20db_16khz.conf"
```

Comment:
16 kHz pipeline should have also the 0 dB gain IIR option

Reply:
I wonder if parameters like cut-off Hz and gain dB could be passed from CMakeLists.txt? This part becomes a bloat when there's many possible parameters to combine. It would also help for next step to calculate filter coefficients in topology compile time.

(just trying to learn about tplg2 - no idea if possible)